plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/LabelEmbedding.kt

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2010-2024 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.formver.embeddings
+
+import org.jetbrains.kotlin.formver.asPosition
+import org.jetbrains.kotlin.formver.embeddings.expression.ExpEmbedding
+import org.jetbrains.kotlin.formver.linearization.LinearizationContext
+import org.jetbrains.kotlin.formver.linearization.pureToViper
+import org.jetbrains.kotlin.formver.viper.MangledName
+import org.jetbrains.kotlin.formver.viper.ast.Label
+import org.jetbrains.kotlin.formver.viper.ast.Stmt
+
+data class LabelLink(val name: MangledName)
+
+data class LabelEmbedding(val labelLink: LabelLink, val invariants: List<ExpEmbedding>? = null) {
+    val name: MangledName
+        get() = labelLink.name
+}
+
+fun LabelLink.toLabel(invariants: List<ExpEmbedding>? = null): LabelEmbedding = LabelEmbedding(this, invariants)
```

Comment:
Is this something we should make available?  What is the use-case?  I'm concerned people will use it thinking it'll find the invariants somewhere.

Reply:
Are you suggesting changing the name to something like `completeWith`?
Or just using the `LabelEmbedding`'s constructor

Reply:
I mean, should we make it easy to start with a `LabelLink` and get a `Label`?  I would think the other direction is the one that should mostly be used.

Reply:
I think we can do it in the other direction. We would need to store these embeddings somewhere in the context.
Here I propose that links are used in gotos and embeddings are used when we add them to the statements.

Reply:
Right, but when do we take a link and create an embedding out of it?